internal/k8shandler/status.go

```diff
@@ -284,76 +284,66 @@ func (er *ElasticsearchRequest) updateNodeConditions(status *api.ElasticsearchSt
+
+	nodeNames := []string{}
+	clusterNodes := nodes[nodeMapKey(er.cluster.GetName(), er.cluster.GetNamespace())]
+
+	for _, node := range clusterNodes {
+		nodeNames = append(nodeNames, node.name())
+	}
+
+	for _, node := range er.cluster.Spec.Nodes {
+		nodeName := nodeNameContains(node.GenUUID, nodeNames)
+
+		if nodeName == "" {
+			ll.Info("Unable to find appropriate node to compare storage spec.")
+			continue
+		}
+
 		specVol := node.Storage
 		current := &v1.PersistentVolumeClaim{}
-		emptySpecVol := api.ElasticsearchStorageSpec{}
+		claimName := fmt.Sprintf("%s-%s", er.cluster.Name, nodeName)
+
+		isUsingPVCStorageSpec := true
+		isEphemeralStorageSpec := reflect.DeepEqual(specVol, emptySpecVol) || specVol.Size == nil
 
-		// When these conditions are not met, ephemeral storage
-		// is used instead of making a pvc claim.
-		didMakePVCClaim := true
-		willMakePVCClaim := !(reflect.DeepEqual(specVol, emptySpecVol) || specVol.Size == nil)
+		if isEphemeralStorageSpec {
+			ll.Info("iS eph")
+		}
 
 		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			if err := client.Get(context.TODO(), types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}, current); err != nil {
+			if err := er.client.Get(context.TODO(), types.NamespacedName{Name: claimName, Namespace: er.cluster.Namespace}, current); err != nil {
 				if apierrors.IsNotFound(err) {
-					didMakePVCClaim = false
+					isUsingPVCStorageSpec = false
+					ll.Info("Not finding claim...")
 				} else {
-					return kverrors.Wrap(err, "failed to get PVC", "claim", cluster.Name)
+					return kverrors.Wrap(err, "failed to get PVC", "claim", claimName)
 				}
 			}
 
-			// No changes and ephemeral storage used, means no status changes
-			if !willMakePVCClaim && !didMakePVCClaim {
+			if isEphemeralStorageSpec && !isUsingPVCStorageSpec {
 				return nil
-			}
-
-			currentSize := current.Spec.Resources.Requests.Storage()
-			structureStatus, nameStatus, sizeStatus := v1.ConditionFalse, v1.ConditionTrue, v1.ConditionTrue
-
-			if willMakePVCClaim != didMakePVCClaim {
+			} else if !isEphemeralStorageSpec != isUsingPVCStorageSpec {
 				structureStatus = v1.ConditionTrue
+				return nil
 			}
 
-			if reflect.DeepEqual(current.Spec.StorageClassName, specVol.StorageClassName) {
-				nameStatus = v1.ConditionFalse
+			if !reflect.DeepEqual(current.Spec.StorageClassName, specVol.StorageClassName) {
+				nameStatus = v1.ConditionTrue
 			}
 
+			currentSize := current.Spec.Resources.Requests.Storage()
 			if currentSize != nil && specVol.Size != nil {
-				if currentSize.Equal(*specVol.Size) {
-					sizeStatus = v1.ConditionFalse
+				if !currentSize.Equal(*specVol.Size) {
+					sizeStatus = v1.ConditionTrue
 				}
-			} else if currentSize == nil && specVol.Size == nil {
-				sizeStatus = v1.ConditionFalse
+			} else if currentSize == nil || specVol.Size == nil {
```

Comment:
since this is the negation of the if statement before it, can this just be an `else`?

Reply:
actually, I meant this to be a different statement. I've just updated it.